### PR TITLE
Condense onChange handlers, add support for age units

### PR DESCRIPTION
**Pull request by doogthemushroom**
### Summary
Reducing the necessary number of onChange handlers from 3 to 1. Also adding a new `ageUnit` prop and onChange value to support initializing correctly with the `age` granularity. Updated some documentation to clarify the Onset Picker does not deal with time.

### Additional Details
To be clear this would be a breaking change, and would require a 4.0 release. I have added notes to the changelog regarding this.

### CODE REVIEW

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx [hunk 1]
```diff
@@ -20,9 +20,15 @@ const GranularityOptions = {
...
 };
 
+const DATE_FORMAT = 'YYYY-MM-DD';
+
 const propTypes = {
   /**
-   * The ISO 8601 string representation of the birth date to calculate an onset date for the 'age' precision.
+   * The date unit of the age value. One of 'weeks', 'months', or 'years'.
+   */
+  ageUnit: PropTypes.string,
```
Comment (benbcai) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Could use PropTypes.oneOf(['weeks', 'months', 'years']).

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx [hunk 2]
```diff
@@ -20,9 +20,15 @@ const GranularityOptions = {
   DATE: 'date',
 };
 
+const DATE_FORMAT = 'YYYY-MM-DD';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: shouldn't this format should be translated

Comment (doogthemushroom) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: No it's for internal use only, it's the date only portion of an ISO-8601 date.

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx (hunk 1 above)
Comment (doogthemushroom) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Updated to ['weeks', 'months', 'years']

Comment (tbiethman) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: I'd just go ahead and do `PropTypes.oneOf(['weeks', 'months', 'years'])`

Comment (doogthemushroom) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Already updated per @benbcai suggestion!

Comment (tbiethman) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: D'oh i'm way out of date here

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx [hunk 3]
```diff
@@ -144,16 +151,14 @@ class OnsetPicker extends React.Component {
...
         };
-      }, () => this.handleDateChange(this.state.onsetDate));
-    }
-    this.setState({ granularity });
-
-    if (this.props.granularitySelectOnChange) {
-      this.props.granularitySelectOnChange(granularity);
+      }, () => this.handleOnsetUpdate());
+    } else {
+      this.setState({ granularity }, () => this.handleOnsetUpdate());
```
Comment (tbiethman) [inline packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: You could remove one function instantiation per execution if you provide handleOnsetUpdate directly as the callback: `this.setState({ granularity }, this.handleOnsetUpdate);`

Comment (doogthemushroom) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Ah of course, updated. I think I like the look of these little anonymous functions too much.

**Comment by emilyrohrbough** [comment]
Are any of these issues/bugs resolved with this pr? 
- https://github.com/cerner/terra-clinical/issues/397
- https://github.com/cerner/terra-clinical/issues/435
- https://github.com/cerner/terra-clinical/issues/461

File: packages/terra-clinical-onset-picker/src/OnsetPicker.jsx (hunk 2 above)
Comment (emilyrohrbough) [reply packages/terra-clinical-onset-picker/src/OnsetPicker.jsx]: Can we add an inline comment to document this use?

**Pull request merged**